src/main/java/core/basesyntax/strategy/handler/PurchaseHandlerImpl.java

```diff
@@ -0,0 +1,22 @@
+package core.basesyntax.strategy.handler;
+
+import core.basesyntax.dao.StorageDao;
+import core.basesyntax.model.Fruit;
+import core.basesyntax.model.FruitTransaction;
+
+public class PurchaseHandlerImpl implements TransactionHandler {
+    private final StorageDao storageDao;
+
+    public PurchaseHandlerImpl(StorageDao storageDao) {
+        this.storageDao = storageDao;
+    }
+
+    @Override
+    public void makeTransaction(FruitTransaction transaction) {
+        Fruit fruit = new Fruit(transaction.getFruit());
+        Integer amount = storageDao.getAmount(fruit);
+        if (storageDao.getAmount(fruit) > transaction.getAmount()) {
+            storageDao.update(fruit,amount - transaction.getAmount());
```

Comment:
Would be better to put such expression inside StorageDaoImpl in update method, so that you can modify code flexible 
```suggestion
        if (storageDao.getAmount(fruit) > transaction.getAmount()) {
            storageDao.update(fruit,amount - transaction.getAmount());
```